doc/_extensions/zephyr/domain/__init__.py

```diff
@@ -797,14 +798,53 @@ def run(self):
             return result_nodes
 
         # Add the note before any tables
-        note = nodes.admonition()
-        note += nodes.title(text="Note")
-        note["classes"].append("note")
-        note += nodes.paragraph(
-            text="The tables below were automatically generated using information from the "
-            "Devicetree. They may not be fully representative of all the hardware features "
-            "supported by the board."
+        note = nodes.admonition(classes=["legend"])
+        note += nodes.title(text="Table Legend", classes=["note"])
+
+        legend = nodes.definition_list(classes=["supported-hardware field-list"])
+
+        def add_legend_item(legend_list, term_text, term_classes, definition_text):
+            dl_item = nodes.definition_list_item()
+            dt = nodes.term()
+            dt += nodes.inline("", term_text, classes=term_classes)
+            dd = nodes.definition()
+            dd += nodes.Text(definition_text)
+            dl_item += dt
+            dl_item += dd
+            legend_list += dl_item
+
+        add_legend_item(
+            legend, "on-chip", ["location-chip", "onchip"], "Feature integrated in the SoC."
         )
+        add_legend_item(
+            legend, "on-board", ["location-chip", "onboard"], "Feature present on the board."
+        )
+        add_legend_item(
+            legend,
+            "2",
+            ["count", "okay-count"],
+            "Number of instances of the feature that are present and enabled.",
+        )
+        add_legend_item(
+            legend,
+            "2",
+            ["count", "disabled-count"],
+            "Number of instances of the feature that are present but initially disabled.",
```

Comment:
Nit: 
```suggestion
            "Number of instances of the feature that are available but not enabled.",
```

Reply:
ya I don't mind this change - but I think the challenge is that saying a feature is "available" on the board (albeit initially not enabled) might sometimes be promising things that are not true? some peripherals "available [on the soc] but not enabled" may simply be never usable on whatever tiny board with very limited pinout, right? So I think I was trying to convey that.

Reply:
For the sake of clarity of this table (and what can be explained in the legend) I think that we should be as close possible to what is actually done in boards description. IE, nodes in .dtsi are available (could potentially be enabled and used) and then made enabled in boards description (and then they are usable "Out of the box").
Consideration about limited resources or promises are another concern.

But this is likely and endless discussion.